### ISSUE TITLE
[FEAT]: Allow Defining Actions Access Settings on a Repository

### ISSUE THREAD

**Issue by cwthrivent** [integrations/terraform-provider-github#1418]
### Describe the need

In a repository, you can go to `Settings -> Actions -> General` and then there is a section for controlling the visibility/scope of the repository with respect to GitHub Actions. Here is a screenshot:

![image](https://user-images.githubusercontent.com/68966638/206579862-ba7c6692-6445-42c0-8c31-701721bb6cd9.png)

Currently, there does not seem to be a way to configure this via this Terraform provider (I apologize in advance if there is).

I found in the GitHub documentation that there is support for some API calls to fetch and set this value:
* `GET` - https://docs.github.com/en/enterprise-cloud@latest/rest/actions/permissions?apiVersion=2022-11-28#get-the-level-of-access-for-workflows-outside-of-the-repository
* `PUT` - https://docs.github.com/en/enterprise-cloud@latest/rest/actions/permissions?apiVersion=2022-11-28#set-the-level-of-access-for-workflows-outside-of-the-repository

These are documented in the context of the GitHub Enterprise API, so I hope they are not out of scope for this provider. Let me know if they are.

### API Version

GitHub REST API 2022-11-28

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kfcampbell** [comment]
Ooh, you're right! This would be a great feature to implement in the provider. I've added the "Status: Up for grabs" label and I'd be happy to review PRs for this behavior.

**Comment by wwsean08** [comment]
I'm curious, I feel like it would make sense for this to be added to the `github_actions_repository_permissions`, does that make sense to you or should it be it's own resource?  This would be modifying settings on the same page which is why I think that it should be added to that resource, but it is slightly different in terms of settings, essentially what repos can reference this as opposed to the are actions allowed and what actions can this repo access.

Just wanted to check before I open a PR :)

**Comment by kfcampbell** [comment]
@cwthrivent the repository you're looking for is [google/go-github](https://github.com/google/go-github)! That's the module we use to interact with GitHub's REST API.

**Comment by s4nji** [comment]
> Edit: also digging in a bit deeper, it looks like an update to go-github will be required before this can be done, as I'm not seeing any function for manipulating these settings the library.

@wwsean08 This was added in the recent release ([v48.2.0](https://github.com/google/go-github/releases/tag/v48.2.0)) via https://github.com/google/go-github/pull/2578 so it should be ready to be added into terraform-provider-github 😄

**Comment by kfcampbell** [comment]
The project has been updated to use v48.2.0 of google/go-github, so this feature is now unblocked!

**Comment by wwsean08** [comment]
Gotcha, I must have been looking at an older version of the source code when I was searching for the comment with the documentation URL.  I can try taking a stab at this on Friday, but curious about your opinion on my proposition of adding this to the `github_actions_repository_permissions` resource @kfcampbell, the things modified by that resource are on the same page as what this would be modifying in the settings UI, otherwise I can learn how to create a new resource from scratch and this may take a bit longer (assuming someone doesn't beat me to it)

**Comment by kfcampbell** [comment]
I'm :+1: on keeping the new functionality in the existing `github_actions_repository_permissions` resource!
